app/models/article.rb

```diff
@@ -262,6 +262,11 @@ def index_or_remove_from_index_where_appropriate
     end
   end
 
+  def webhook_data
```

Comment:
I agree that it's not an article's responsibility but I added this method so that we could implement `webhook_data` for other objects so that `DispatchEvent` wouldn't need to care for this. (though for now we only need articles :thinking: )
Do you suggest creating another class to find out the needed serializer and serialize?

Reply:
> I agree that it's not an article's responsibility but I added this method so that we could implement webhook_data for other objects so that DispatchEvent wouldn't need to care for this

I understand your thinking, I think it would be neater to have the code as centralized as possible, that doesn't mean a single class has to do everything but the less the code is scattered the easier it should be to maintain and in this case the responsibility is quite clear in my opinion, it shouldn't be the model responsibility how to serialize itself for webhooks, the same way we don't have the JSON serialization code of the APIs inside the models

Reply:
So where would you put the "choosing serializer" code for now?

Reply:
The code structure could be different if we switch to oj though.

Reply:
@lightalloy oj is used transparently by fast api, no change needed. 

> So where would you put the "choosing serializer" code for now?

In an adapter, and inside `DispatchEvent` you could call it with something like:

```ruby
event_json = Event.new(event_type: event_type, payload: Webhooks::PayloadAdapter.call(object)).to_json
```

or you could create an includeable model that you include inside `class Article` if you want to keep `object.webhook_data`

We keep adding stuff to the model and this seems an occasion not to do it :D

Reply:
> We keep adding stuff to the model and this seems an occasion not to do it :D
Yeah, this looks like a fake refactoring in my opinion because stuff is still in the model :)